### ISSUE TITLE
Hamburger menu like on the docs website

### ISSUE THREAD

**Issue by Odonno** [picturepan2/spectre#415]
The Hamburger Menu on the docs website is pretty interesting.

* supports responsive design
* full height on the sidebar on large screen
* hamburger menu button (+ overlay sidebar) on small screen

Really interested to see it in this library.

**Comment by Janca** [comment]
Its already supported with the experimental CSS. Its a mix of a few different components, and clever pseudo-selectors like `':target'`. This example below may help get you started.

```html
<div class="off-canvas off-canvas-sidebar-show">
    <div id="left-navigation" class="off-canvas-sidebar">
        <div class="navigation-menu">
            <a href="/" class="logo">
                <img src="/assets/brand.png" class="brand">
            </a>
            <div class="nav-menus">
                <details class="accordion">
                    <summary class="accordion-header c-hand">
                        <span class="nav-item header">Main Nav Item 1</span>
                    </summary>
                    <div class="accordion-body">
                        <ul class="menu menu-nav">
                            <li class="menu-item">
                                <a href="#" class="nav-item">Main Nav Item 1 Sub Item 1</a>
                            </li>
                            <li class="menu-item">
                                <a href="#" class="nav-item">Main Nav Item 1 Sub Item 2</a>
                            </li>
                            <li class="menu-item">
                                <a href="#" class="nav-item">Main Nav Item 1 Sub Item 2</a>
                            </li>
                        </ul>
                    </div>
                </details>
                <details class="accordion">
                    <summary class="accordion-header c-hand">
                        <span class="nav-item header">Main Nav Item 2</span>
                    </summary>
                    <div class="accordion-body">
                        <ul class="menu menu-nav">
                            <li class="menu-item">
                                <a href="#" class="nav-item">Main Nav Item 2 Sub Item 1</a>
                            </li>
                            <li class="menu-item">
                                <a href="#" class="nav-item">Main Nav Item 2 Sub Item 2</a>
                            </li>
                            <li class="menu-item">
                                <a href="#" class="nav-item">Main Nav Item 2 Sub Item 2</a>
                            </li>
                        </ul>
                    </div>
                </details>
                <!-- ETC -->
            </div>
        </div>
    </div>

    <a class="off-canvas-overlay" href="#close"></a>

    <div class="off-canvas-content">
        <div id="main-content">
            <!-- THIS IS WHERE YOUR MAIN CONTENT WOULD GO -->
        </div>
    </div>
</div>
```

And there is a little bit of custom CSS that the doc's page applies that I think the LESS below will mimic fairly accurately.
```less
.nav-menus {
    margin-top: 12px;
    padding: 0 .75rem;

    .nav-item {
      &.header {
        text-transform: uppercase;
        font-weight: 600;
      }

      font-size:0.7rem;
    }
  }
```

**Comment by Janca** [comment]
I am not sure that it belongs as a component on the documentation as it is more of a collection of components rather than a single one. It would make a good example if there were an example section. Perhaps make a PR for an example page including the off-canvas navigation bar as well as a few other components that could work well together.

**Comment by edceliz** [comment]
An example page is really a good idea. There should be a page for Spectre usage such as templates and widgets.

On that note, a new issue was opened related to this
> Issue https://github.com/picturepan2/spectre/issues/416

**Comment by Odonno** [comment]
@Janca Examples are a really good idea. But the point here is to be able to copy/paste code and be able to create an HamburgerMenu in a few seconds. Something that can be found on the Experimentals page for example.

If you think it is a different category, we can create another page/category.

**Comment by Teejay1589** [comment]
I agree with Janca about having a page for templates
I just came across SPECTRE CSS and I love to use it soo much I am thinking of making templates with it.

Concerning the OFF CANVAS Experimental Feature, I had to use a little additional CSS & jQuery to make it look exactly like that in the DOCUMENTATION.

**Comment by Teejay1589** [comment]
I agree with Janca about having a page for templates
I just came across SPECTRE CSS and I love to use it soo much I am thinking of making templates with it.

Concerning the OFF CANVAS Experimental Feature, I had to use a little additional CSS & jQuery to make it look exactly like that in the DOCUMENTATION.

**Comment by Janca** [comment]
@Teejay1589 I was able to replicate the documentation's navigation menu with only the additional classes that I included in my example post. Do you remember where you had to use jQuery to achieve your styling?

On topic though, I believe that while copy-and-paste snippets would be extremely useful, the way that the off-canvas classes are designed you almost need to style your entire page around using it. It isn't just a copy-pasta kinda thing. An example page using the off-canvas toolbar while providing code blocks for how its used on the page would be better. Otherwise you would be no better off than just linking to the current documentation on [off-canvas](https://picturepan2.github.io/spectre/experimentals.html#off-canvas).

**Comment by Odonno** [comment]
So, I suppose we can organize in that way:

* let the current demonstration/components as is
* add a Templates section (with some code to get started with a HamburgerMenu design, etc..)
* add a Examples section (with complete examples of living websites)

**Comment by Janca** [comment]
I agree with keeping the current code examples as well as the experimental examples the same. I would then create a new category, perhaps 'Advanced Examples', and include in-depth code examples of using multiple Spectre components together to form larger components. I have no other opinions on what should be included in this category.

Also, as I have said before, creating an example of this off-canvas navigation menu will not be a one-to-one visual comparison as there are other styles applied to the navigation on the documentation pages. Maybe it would be better to only include this 'Advanced Examples' section when the appropriate style classes have been made, or one could maybe include it in their PR for the new documentation pages.
